File: javascript/gemini/p108.js
const { exec } = require('child_process');

const username = 'root';

// exec handles the shell execution and returns the output in a callback
const output = exec(`getent passwd ${username}`, (error, stdout, stderr) => {
    if (error) {
        console.error(`Error: ${error.message}`);
        return;
    }
    if (stderr) {
        console.error(`Stderr: ${stderr}`);
        return;
    }
    console.log(`User Info: ${stdout}`);
});

File: java/gemini/p108.java
import java.io.*;

public class GetEnt {
    public static void main(String[] args) {
        String username = "root";
        try {
            // Executing the command
            Process process = Runtime.getRuntime().exec("getent passwd " + username);
            
            // Getting the stream (equivalent to the output pointer)
            BufferedReader output = new BufferedReader(
                new InputStreamReader(process.getInputStream())
            );

            String line;
            while ((line = output.readLine()) != null) {
                System.out.println(line);
            }
            output.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}